### ISSUE TITLE
Chamfer Distance Norm

### ISSUE THREAD

**Issue by Seanmvcollins** [NVlabs/DigitalTwinArt#3]
Hi, 

Looking at your code-base, I came across an error which I think you will want to know about.

In your paper it claims that L1 Chamfer Distance is used: "We evaluate object and part mesh reconstructions with bi-directional Chamfer-l1 distance (CD)". However, I have had a look at where you use PyTorch3D's chamfer_distance() function, in [eval/eval_mesh.py](https://github.com/NVlabs/DigitalTwinArt/blob/master/eval/eval_mesh.py#L31), and you are not specifying the norm to be 1. By default, the norm for chamfer_distance() is 2, which can be seen in the [PyTorch3D docs](https://pytorch3d.readthedocs.io/en/latest/modules/loss.html), as well as the [PyTorch3D 0.7.4 source code](https://github.com/facebookresearch/pytorch3d/blob/v0.7.4/pytorch3d/loss/chamfer.py). When I modify the code to specify the norm to equal the default, that is 2, and train on some of the object categories, I am able to reproduce results from your Table 1. However, if I specify the norm to equal 1, I get results significantly higher.

Results for (true) L1 Chamfer Distance - Blade
![Blade_Results_True_L1](https://github.com/NVlabs/DigitalTwinArt/assets/71595099/4e7d88a3-5a94-4439-a1b4-3ba4990a7234)

Results for L2 Chamfer Distance - Blade
![Blade_Results_L2](https://github.com/NVlabs/DigitalTwinArt/assets/71595099/5ab45eff-c76b-4245-afac-e4c11feec510)

Comparing Blade L2 Chamfer Distance with Table 1 results
![Replication](https://github.com/NVlabs/DigitalTwinArt/assets/71595099/04ba7350-f43d-41ce-8648-33c32c751364)

I am fairly sure that this error originates with the [PARIS code-base](https://github.com/3dlg-hcvc/paris).

Good day,
Sean

**Comment by HalfSummer11** [comment]
Thanks for raising this important question, Sean! It is not intuitive, but "L1" here refers to the fact that it's measuring the distance (mm) instead of the squared distance (mm^2). The [definition](https://www.cvlibs.net/publications/Mescheder2019CVPR_supplementary.pdf) and [implementation](https://github.com/autonomousvision/occupancy_networks/blob/406f79468fb8b57b3e76816aaa73b1915c53ad22/im2mesh/eval.py#L71) of "Chamfer-L1" and "Chamfer-L2" can be traced back to Occupancy Networks (maybe earlier).

I agree that metrics are very important and it can be confusing when works use different metrics/implement them differently. In the codebases of A-SDF and Ditto which followed A-SDF, the stated "Chamfer-L1" is in fact implemented as "Chamfer-L2", i.e. the average of squared distances. We followed the most recent related work PARIS for both the description and implementation of the metric. We also ensured all methods (except Ditto which we took from PARIS' table) were evaluated using the same protocol.
